aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -298,8 +298,13 @@ private List<String> deleteBatch(String bucket, Collection<String> keysToDelete)
   @Override
   public Iterable<FileInfo> listPrefix(String prefix) {
     S3URI s3uri = new S3URI(prefix, s3FileIOProperties.bucketToAccessPointMapping());
+    String suffix = "";
+    if (S3Express.checkIfS3Express(s3uri.bucket()) && !prefix.endsWith("/")) {
```

Comment:
you could think about doing this everywhere.
Good: consistent behaviour, better test coverage
Bad: if there is a regression, it'll surface more broadly.

personally I'd do it everywhere...

Reply:
What do you mean by everywhere Steve?
What other places could I cover?

(This is my first pr to iceberg so I would like to be as safe as possible.)